main.go

```diff
@@ -0,0 +1,55 @@
+// grep | (c) 2020 NETWAYS GmbH | GPLv2+
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+)
+
+func main() {
+	patternParameter := flag.String("e", "", "use PATTERN for matching.")
+	flag.Parse()
+
+	if *patternParameter != "" {
+		re, regErr := regexp.Compile(*patternParameter)
+		if regErr != nil {
+			fmt.Fprintln(os.Stderr, regErr)
+			os.Exit(1)
+		}
+
+		buf := bufio.NewReader(os.Stdin)
+
+		for {
+			data, dataErr := buf.ReadBytes('\n')
+			if dataErr != nil && dataErr != io.EOF {
+				fmt.Fprintln(os.Stderr, dataErr)
+				os.Exit(1)
+			}
+
+			if len(data) == 0 {
+				break
+			}
+
+			if !bytes.Contains(data, []byte{'\n'}) {
+				data = append(data, '\n')
+			}
+
+			if re.Match(data) {
+				_, _ = os.Stdout.Write(data)
+			}
+
+			if dataErr == io.EOF {
+				break
+			}
+		}
+	} else {
+		fmt.Fprintln(os.Stderr, "grep: at least one parameter is needed.")
```

Comment:
"at least one" of **which** parameter? Yes, at the moment there's only one, but have a look at the issue list.

Reply:
you can see the parameters with grep --help or in my case go run . --help

Reply:
Yes, I see there **what's available**, but in the error message I don't see **what I missed**.

Reply:
The message was almost right, you had just to add that it's about the **-e** parameter.